### PR TITLE
Remove `Net5CompatFileStreamStrategy`.

### LINKED ISSUE #55196: Remove Net5CompatFileStreamStrategy in .NET 7 early previews
cc @stephentoub

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
Fixes #55196. I leave it here to be merged when seen fit.

c.c. @adamsitnik

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.Windows.cs
Comment (teo-tsirpanis) [inline src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.Windows.cs:19]: This class is unused but I didn't remove it to avoid surprising any new code that might decide to use it in the meantime until the PR is merged.

**Review by adamsitnik — APPROVED** [review]
"sight for sore eyes", thank you @teo-tsirpanis !

**Comment by teo-tsirpanis** [comment]
It took me one Google search to understand what you said (thought there was a problem with your eyes and was confused), but I'm glad you liked it. 😊 Removing old code is satisfactory for both the doer and the beholder.

**Comment by jeffhandley** [comment]
@adamsitnik Our plan was to do this in early 7.0 previews, but after we were sure 6.0 was stabilized and on its path toward GA. If we merge this eagerly, then any changes/fixes to the file stream strategy work might have conflicts between `main` and the `release/6.0` branch(es). But I'll leave it to your discretion whether or not there's any risk in that. Worst case, we'd have to make a fix directly into `release/6.0` that doesn't land in `main` first.

**Comment by eerhardt** [comment]
@teo-tsirpanis - can you resolve the conflicts in this PR?

**Comment by teo-tsirpanis** [comment]
Done @eerhardt.

**Comment by Jozkee** [comment]
Breaking change doc issue: https://github.com/dotnet/docs/issues/31465

**Pull request merged**
